### ISSUE TITLE
Constant bpf map op failures on nodeport_neigh4 in GKE

### ISSUE THREAD

**Issue by dctrwatson** [cilium/cilium#15381]
## Bug report

the [new periodic neighbor refresh](https://github.com/cilium/cilium/pull/14498) causes constant bpf map op failures on `nodeport_neigh4`

There are no logs about the error, just metrics.

If I turn on debug, most of the logging is just Register/Unregister that map every ~100ms. If I set `net.ipv4.neigh.default.base_reachable_time_ms` to some long interval, then it only happens for 180-240s at the interval I set the sysctl to.

**General Information**

- Cilium version (run `cilium version`): v1.8.7, v1.8.8
- Kernel version (run `uname -a`): 4.19.150+
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...): v1.17.17-gke.1100

**Comment by brb** [comment]
@dctrwatson Can you include the full metric and also the debug log msgs?

**Comment by brb** [comment]
OK, I see one problem that each time we retire a neighbor entry from the BPF map, we open/close the map: https://github.com/cilium/cilium/blob/master/pkg/maps/neighborsmap/neighborsmap.go#L146. This definitely needs a fix.

However, I still need the metric name to see which op is failing.

**Comment by dctrwatson** [comment]
Debug logs:
```
2021-03-17T05:28:16.884496496Z level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
2021-03-17T05:28:16.884554710Z level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
2021-03-17T05:28:17.285488410Z level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
2021-03-17T05:28:17.285525378Z level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
2021-03-17T05:28:17.294407459Z level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
2021-03-17T05:28:17.294434297Z level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
2021-03-17T05:28:17.868475477Z level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
2021-03-17T05:28:17.868512159Z level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
2021-03-17T05:28:18.307490746Z level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
```

Metric:
`cilium_bpf_map_ops_total                              mapName="nodeport_neigh4" operation="delete" outcome="fail"                                       233.000000`

**Comment by borkmann** [comment]
Should be fixed in latest image (1.11).. pls try `quay.io/cilium/cilium-ci:latest` and if it still doesn't work, pls reopen. Thanks!
